backend/src/apiserver/main.go

```diff
@@ -72,12 +80,47 @@ func main() {
 		template.Launcher = common.GetStringConfig(launcherEnv)
 	}
 
-	clientManager := cm.NewClientManager()
+	logLevel := *logLevelFlag
+	if logLevel == "" {
+		logLevel = "info"
+	}
+
+	var zapLevel zapcore.Level
+	switch logLevel {
+	case "debug":
+		zapLevel = zapcore.DebugLevel
+	case "info":
+		zapLevel = zapcore.InfoLevel
+	case "warn":
+		zapLevel = zapcore.WarnLevel
+	case "error":
+		zapLevel = zapcore.ErrorLevel
+	case "dpanic":
+		zapLevel = zapcore.DPanicLevel
+	case "panic":
+		zapLevel = zapcore.PanicLevel
+	case "fatal":
+		zapLevel = zapcore.FatalLevel
+	default:
+		glog.Fatalf("Invalid log level: %s", logLevel)
+	}
+
+	ctrllog.SetLogger(zap.New(zap.UseFlagOptions(&zap.Options{Level: zapLevel})))
+
+	clientManager, err := cm.NewClientManager()
+	if err != nil {
+		glog.Fatalf("Failed to initialize ClientManager: %v", err)
+	}
+	if common.IsOnlyKubernetesWebhookMode() {
+		startWebhookHTTPSProxy(clientManager.ControllerClient())
+		clientManager.Close()
```

Comment:
Should this be a defered function instead after the `clientManager` error is checked?